portfolio/src/main/webapp/script.js

```diff
@@ -46,18 +46,31 @@ function addRandomQuote() {
 }
 
 function getComments() { //examples/stats-server
-    fetch('/data').then(response => response.json()).then((strings) => {
-        const commentListElement = document.getElementById('comment-container');
-        commentListElement.innerHTML = '';
-        strings.forEach((line) => {
-            commentListElement.appendChild(createListElement(line));
-        });
+    var loggedIn = false;
+    fetch('/login').then(response => response.json()).then((logs) =>{
+        loggedIn = logs[0];
+        if (loggedIn){
+            fetch('/data').then(response => response.json()).then((strings) => {
```

Comment:
According to my Chrome debugger trace, this is where we get "Uncaught (in promise) SyntaxError: Unexpected token < in JSON at position 0", and I thought that's because the response returned from your DataServlet.java (which serves the `/data` URL path) might return a non-JSON output, but probably some HTML content (highly suspicious because it says `Unexpected token < in JSON at position 0`) is returned. However, I checked your DataServlet.java, and it does return a JSON response, so I'm clueless at this point...

It looks like the double `fetch`-ing is not a problem (my debugger trace was fine with that), and I'm clueless about the JSON syntax error, so I'd recommend you ask this question at the Tech/Project chat to other more experienced web developer Googlers.

I'll keep looking myself, but note that it may take a while...

Reply:
Found the reason why this throws a JSON syntax error. That's because the last line in your `doGet()` of your DataServlet.java is a redirect statement, which shouldn't be there in the first place. After removing that line, I got your comments all showing on my devserver Preview of your code!